core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -446,6 +448,27 @@ protected User doAddUserWithID(String userName, Object credential, String[] role
         throw new NotImplementedException("doAddUserWithID operation is not implemented in: " + this.getClass());
     }
 
+    /**
+     * Update claims that contain invalid user ID with generated user ID.
+     *
+     * @param userID new user ID of the user.
+     * @param claims claim map for the user.
+     */
+    protected void updateClaimsWithUserId(String userID, Map<String, String> claims) {
```

Comment:
As this method specifically handle only the LOCATION_CLAIM_URI shall we rename the method accordingly?

Reply:
Fixed in 494db58